File: RentTheGown/RentTheGown/ViewController.m
//
//  ViewController.m
//  RentTheGown
//
//  Created by Billy Tobon on 2/15/15.
//  Copyright (c) 2015 Rent The Runway. All rights reserved.
//

#import "ViewController.h"
#import "RENTButton.h"


@interface ViewController ()

@end

@implementation ViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    RENTButton *aButton = [RENTButton buttonWithSKU:@"DRESS001" title:@"Rent me!" completionBlock:^(BOOL completed){
        
        UIAlertView *alert = [[UIAlertView alloc] initWithTitle:@"Alright!" message:@"Your dress is on the way" delegate:nil cancelButtonTitle:@"OK" otherButtonTitles:nil];
        [alert show];
        
    }];
    
    aButton.translatesAutoresizingMaskIntoConstraints = NO;
    
    
    [self.view addSubview:aButton];
    
    [self.view addConstraint:[NSLayoutConstraint constraintWithItem:aButton
                                                          attribute:NSLayoutAttributeCenterX
                                                          relatedBy:NSLayoutRelationEqual
                                                             toItem:self.view
                                                          attribute:NSLayoutAttributeCenterX
                                                         multiplier:1.0
                                                           constant:0.0]];
    
    [self.view addConstraint:[NSLayoutConstraint constraintWithItem:aButton
                                                          attribute:NSLayoutAttributeBottom
                                                          relatedBy:NSLayoutRelationLessThanOrEqual
                                                             toItem:self.view
                                                          attribute:NSLayoutAttributeBottom
                                                         multiplier:1.0
                                                           constant:-40.0]];
    

    [self.view addConstraint:[NSLayoutConstraint constraintWithItem:aButton
                                                          attribute:NSLayoutAttributeWidth
                                                          relatedBy:NSLayoutRelationEqual
                                                             toItem:nil
                                                          attribute:NSLayoutAttributeNotAnAttribute
                                                         multiplier:1.0
                                                           constant:150.0]];
    
    [self.view addConstraint:[NSLayoutConstraint constraintWithItem:aButton
                                                          attribute:NSLayoutAttributeHeight
                                                          relatedBy:NSLayoutRelationEqual
                                                             toItem:nil
                                                          attribute:NSLayoutAttributeNotAnAttribute
                                                         multiplier:1.0
                                                           constant:40.0]];
    
}

- (void)didReceiveMemoryWarning {
    [super didReceiveMemoryWarning];
}

@end

File: RentACorgi/RentACorgi/ViewController.swift
//
//  ViewController.swift
//  RentACorgi
//
//  Created by Billy Tobon on 5/22/15.
//  Copyright (c) 2015 Rent the Runway. All rights reserved.
//

import UIKit
import RentKit

class ViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        // Do any additional setup after loading the view, typically from a nib.
        
        var rentButton = RENTButton(SKU: "CORGI001", title: "Rent me") { (Bool) -> Void in
            
            var alert = UIAlertController(title: "Woof!", message: "You got Rosco for one hour tomorrow", preferredStyle: UIAlertControllerStyle.Alert)
            alert.addAction(UIAlertAction(title: "OK", style: UIAlertActionStyle.Default, handler: nil))
            self.presentViewController(alert, animated: true, completion: nil)
            
            
        }
        
        rentButton.backgroundColor = UIColor.clearColor()
        rentButton.setTranslatesAutoresizingMaskIntoConstraints(false)
        
        view.addSubview(rentButton)
        
        let constraintCenterX = NSLayoutConstraint(item: rentButton, attribute: .CenterX, relatedBy: .Equal, toItem: view, attribute: .CenterX, multiplier: 1, constant: 0)
        
        view.addConstraint(constraintCenterX)
        
        let constraintBottom = NSLayoutConstraint(item: rentButton, attribute: .Bottom, relatedBy: .LessThanOrEqual, toItem: view, attribute: .Bottom, multiplier: 1, constant: -40.0)
        
        view.addConstraint(constraintBottom)
        
        let constraintWidth = NSLayoutConstraint(item: rentButton, attribute: .Width, relatedBy: .Equal, toItem:nil, attribute: .Width, multiplier: 1, constant: 150.0)
        
        view.addConstraint(constraintWidth)
        
        let constraintHeight = NSLayoutConstraint(item: rentButton, attribute: .Height, relatedBy: .Equal, toItem:nil, attribute: .Height, multiplier: 1, constant: 40.0)
        
        view.addConstraint(constraintHeight)
        
        
    }

    override func didReceiveMemoryWarning() {
        super.didReceiveMemoryWarning()
        // Dispose of any resources that can be recreated.
    }


}

